### DISCUSSION TITLE
How do I restore a backup when I have deployed basically anything with ArgoCD?

### DISCUSSION THREAD

**Discussion by TheAnachronism** [longhorn/longhorn#4857] [Q&A] [answered]
The problem I have is, even if I tell ArgoCD to NOT automatically sync deployments, I don't know how I can tell the current deployment to now use the restored version.
More specifically, I have Gitea deployed with ArgoCD and Helm. The helm chart of Gitea uses an STS for both the postgres database and Gitea itself. Within the STS is the PVC configuration, which means, as soon as I delete the pvc it gets recreated.

Because of that, I don't know how I can tell K8S or Longhorn, that the data of a restored volume should be used, instead of creating a new one.

Which steps should I follow, to restore a backup made by longhorn?

**Comment by shuo-wu** [comment]
In brief, your workload pod can use a fixed PVC or use a PVC with a fixed name, right?
If YES, I think there are 2 possible workarounds:
1. Bind a restored Longhorn volume with an existing PVC: You need to remove the PV and the newly created Longhorn volume for the existing PVC, then create a PV with the same name for the restored volume. Finally, the existing PVC will be bound with the recreated PV.
2. Create a PVC with a specified name for a restored volume: This is pretty simple, you can directly do it via UI.

I haven't verified the above methods. But the basic idea is that you can customize the PVC and PV for any Longhorn volumes by yourself.

**Comment by TheAnachronism** [comment, marked as answer]
OK, so I solved it with a different way. I scaled down the pods of the services I wanted to restore to **zero**. With that, I recreated the PVC with the correct name from the backup, and then scaled them back up.
During that entire thing, I also disabled auto-sync in ArgoCD. This way I could delete the _new_ but wrong PVCs and replace them with those from the backup.
